test/base-type.cpp

```diff
@@ -44,4 +44,24 @@ BOOST_AUTO_TEST_CASE(instantiate)
 	BOOST_CHECK(p);
 }
 
+BOOST_AUTO_TEST_CASE(sort_by_load_after)
+{
+	auto types (Type::GetConfigTypesSortedByLoadDependencies());
+
+	std::unordered_set<Type*> previousTypes;
+	for (auto type : types) {
+		BOOST_CHECK_EQUAL(true, ConfigObject::TypeInstance->IsAssignableFrom(type));
+
+		if (previousTypes.size() == 0) {
+			BOOST_CHECK_EQUAL(0, type->GetLoadDependencies().size());
+		} else {
+			for (Type* dependency : type->GetLoadDependencies()) {
+				BOOST_CHECK_EQUAL(true, previousTypes.find(dependency) != previousTypes.end());
```

Comment:
That should probably use one of the check macros where you can specify a message with extra context. To find out what will happen, I added `load_after Service` to `host.ti`, and all I got was this not very helpful message:

```
error: in "base_type/sort_by_load_after": check true == previousTypes.find(dependency) != previousTypes.end() has failed [true != false]
```